primitives/core/src/offchain/testing.rs

```diff
@@ -156,10 +158,12 @@ impl OffchainState {
 	}
 
 	fn fulfill_expected(&mut self, id: u16) {
-		if let Some(mut req) = self.expected_requests.remove(&RequestId(id)) {
-			let response = req.response.take().expect("Response checked while added.");
+		if let Some(mut req) = self.expected_requests.remove(&RequestId(self.request_ctr)) {
```

Comment:
should the `remove` not be on `id`?

Reply:
The `id` is the "real" request id (comes from outside the testing helpers) and is calculated from the length of the `requests` collection, which is going to stay at 0 when requests are executed sequentially. The fact that `id` doesn't work here is sort of where the bug/quirk is.